java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -481,26 +445,40 @@ static BufferChunk findSingleRead(BufferChunk first) {
    * @param zcr the zero copy shim
    * @param list the disk ranges within the file to read
    * @param doForceDirect allocate direct buffers
+   * @param minSeekSize the minimum gap to prefer seek vs read
+   * @param minSeekSizeTolerance allowed tolerance for extra bytes in memory as a result of
+   *                             minSeekSize
    */
   static void readDiskRanges(FSDataInputStream file,
                              HadoopShims.ZeroCopyReaderShim zcr,
                              BufferChunkList list,
-                             boolean doForceDirect) throws IOException {
+                             boolean doForceDirect,
+                             int minSeekSize,
+                             double minSeekSizeTolerance) throws IOException {
     BufferChunk current = list == null ? null : list.get();
     while (current != null) {
       while (current.hasData()) {
         current = (BufferChunk) current.next;
       }
-      BufferChunk last = findSingleRead(current);
       if (zcr != null) {
+        BufferChunk last = findSingleRead(current);
         zeroCopyReadRanges(file, zcr, current, last, doForceDirect);
+        current = (BufferChunk) last.next;
       } else {
-        readRanges(file, current, last, doForceDirect);
+        ChunkReader chunkReader = ChunkReader.create(current, minSeekSize);
+        chunkReader.readRanges(file, doForceDirect, minSeekSizeTolerance);
+        current = (BufferChunk) chunkReader.to.next;
       }
-      current = (BufferChunk) last.next;
     }
   }
 
+  static void readDiskRanges(FSDataInputStream file,
```

Comment:
ditto. We should put this public method after the method description.

Reply:
moved the package local method as suggested